react.md

```diff
@@ -4,87 +4,98 @@ Try your best to answer each question on your own before looking up the answer o
-- React is a modern, efficient answer to complex UI applications
-- React is a full stack framework for modern web applications
-- React is a flexible library that plays the role of V in an MVC framework
+- React was created to be simple, so that even people with minimal code experience could use it and create Single Page Applications (SPAs) - False
+
+- React is a modern, efficient answer to complex UI applications - TRUE
+
+- React is a full stack framework for modern web applications - TRUE
+
+- React is a flexible library that plays the role of V in an MVC framework-FALSE
+
 
- 
  #### 2. What are "smart"(logic) and "dumb"(display) components? Explain the difference and also add why we bother to make the distinction between them.
- 
- 
+
+
  //Your Answer
- 
- 
+ <!-- "smart"(logic) components contain all functions and doing the meat of the work while "dumb"(display) components basically just pass through or call on something simple. For the dumb component we just need it to show up on the screen, the smart one will tell everything what to do.
+  -->
+
  //Googled Answer
- 
- 
+ Now, your dumb components should contain only enough javascript/react code in order to work in your app. The dumb components' single responsibility is to control the HTML composition, or presentation, of the component. This is why some people call it a presentation component. To continue keeping things simple: your smart components should do everything else that is related to that specific component, and the specific responsibility of that component, that is not related to the actual HTML of that component. This includes things like:  Mapping state to the properties of the presentation component
+ Mapping action dispatchers to the properties of the presentation component
+ Creating any custom state, event handlers, or other muck.
+ Other business logic for the component that is needed to get it to respond and interact appropriately
+ Whatever other code is needed to ensure that a proper 'viewModel', or finite set of properties, is passed to the presentation component in order for that presentational component to display and function correctly.
+ To summarize, we should have business logic in one file (smart components) that generate a single object (props) that is passed to their HTML template counterpart (dumb component) in order to render a fully functioning component.
+
 #### 3. When we use "yarn add ..." in the terminal - what is yarn doing? And what file will always be automatically updated after we add a package with yarn?
- 
- 
+
+
  //Your Answer
- 
- 
+ yarn add... creates and runs a reaction file. It is updating the browser.
+
  //Googled Answer
- 
- 
+ In general, a package is simply a folder with code and a package.json file that describes the contents. When you want to use another package, you first need to add it to your dependencies. This means running yarn add [package-name] to install it into your project.
```

Comment:
You found a lot of great information in your research! It may help to remember that yarn is something called a package manager.